pylint/lint/expand_modules.py

```diff
@@ -148,13 +150,12 @@ def expand_modules(
                     subfilepath, is_namespace, path=additional_search_path
                 )
                 submodname = ".".join(modpath)
-                result.append(
-                    {
-                        "path": subfilepath,
-                        "name": submodname,
-                        "isarg": False,
-                        "basepath": filepath,
-                        "basename": modname,
-                    }
-                )
+                result[subfilepath] = {
+                    "path": subfilepath,
+                    "name": submodname,
+                    # Preserve arg flag if module is also explicitly specified.
+                    "isarg": result.get(subfilepath, {}).get("isarg", False),  # type: ignore[call-overload]
+                    "basepath": filepath,
+                    "basename": modname,
+                }
```

Comment:
```suggestion
                # Preserve arg flag if module is also explicitly specified.
                isarg = False if subfilepath not in result else result["subfilepath"].get("isarg", False)
                result[subfilepath] = {
                    "path": subfilepath,
                    "name": submodname,
                    "isarg": isarg,
                    "basepath": filepath,
                    "basename": modname,
                }
```

Reply:
OK. I do not like to appease linters, except when they're actually right. :)
But, I changed it to:
```python
isarg = subfilepath in result and result[subfilepath]["isarg"]
```
which is still fairly compact and has the same effect.